crypto/x509/v3_purp.c

```diff
@@ -117,6 +120,16 @@ int X509_PURPOSE_get_count(void)
     return sk_X509_PURPOSE_num(xptable) + X509_PURPOSE_COUNT;
 }
 
+/* find smallest identifier not yet taken - note there might be gaps */
+int X509_PURPOSE_get_fresh_id(void)
```

Comment:
IMO `X509_PURPOSE_get_unused_id()` would be a better name.

Reply:
changed